base/socket.jl

```diff
@@ -2,6 +2,7 @@
 abstract IPAddr
 
 Base.isless{T<:IPAddr}(a::T, b::T) = isless(a.host, b.host)
+Base.convert{T<:Integer}(dt::Type{T}, ip::IPAddr) = dt(ip.host)
```

Comment:
@pao brilliant - thanks. I'm too tired to grok it entirely right now but I will do my best tomorrow.